graph/src/data/query/query.rs

```diff
@@ -112,19 +112,20 @@ impl serde::ser::Serialize for QueryVariables {
 
 #[derive(Clone, Debug)]
 pub enum QueryTarget {
-    Name(SubgraphName),
-    Deployment(DeploymentHash),
+    Name(SubgraphName, VersionNumber),
+    Deployment(DeploymentHash, VersionNumber),
 }
```

Comment:
I tried to look for the best place to put the requested version and `QueryTarget` won the contest.